curator-recipes/src/test/java/org/apache/curator/framework/recipes/nodes/TestPersistentTtlNode.java

```diff
@@ -187,4 +187,34 @@ public void childEvent(CuratorFramework client, PathChildrenCacheEvent event) th
             assertNull(client.checkExists().forPath("/test"));
         }
     }
+
+    @Test
+    public void testTouchNodeNotCreated() throws Exception {
+        try (CuratorFramework client =
+                CuratorFrameworkFactory.newClient(server.getConnectString(), new RetryOneTime(1))) {
+            client.start();
+            final long ttlMs = 1_000L;
+            try (PersistentTtlNode node = new PersistentTtlNode(client, "/test", ttlMs, new byte[0])) {
+                node.start();
+                assertTrue(node.waitForInitialCreate(timing.session(), TimeUnit.MILLISECONDS));
+                // Give some minor time for touch node to be created. Will worked after patch
```

Comment:
I think we are testing a situation that "touch" node is not scheduled to be created. May be we can create a overrideable method, say "scheduleTouchTask", to use for test.

Reply:
I provided a new implementation with a new approach that is much more accurate in the second commit. Basically I will watch all lifecycle for main and  touch nodes with PersistentWatch